homeassistant/components/light/intent.py

```diff
@@ -26,75 +31,117 @@ async def async_setup_intents(hass: HomeAssistant) -> None:
     intent.async_register(hass, SetIntentHandler())
 
 
-def _test_supports_color(state: State) -> None:
-    """Test if state supports colors."""
-    supported_color_modes = state.attributes.get(ATTR_SUPPORTED_COLOR_MODES)
-    if not color_supported(supported_color_modes):
-        raise intent.IntentHandleError(
-            f"Entity {state.name} does not support changing colors"
-        )
-
-
-def _test_supports_brightness(state: State) -> None:
-    """Test if state supports brightness."""
-    supported_color_modes = state.attributes.get(ATTR_SUPPORTED_COLOR_MODES)
-    if not brightness_supported(supported_color_modes):
-        raise intent.IntentHandleError(
-            f"Entity {state.name} does not support changing brightness"
-        )
-
-
-class SetIntentHandler(intent.IntentHandler):
+class SetIntentHandler(intent.ServiceIntentHandler):
     """Handle set color intents."""
 
     intent_type = INTENT_SET
     slot_schema = {
-        vol.Required("name"): cv.string,
+        **intent.ServiceIntentHandler.slot_schema,
         vol.Optional("color"): color_util.color_name_to_rgb,
         vol.Optional("brightness"): vol.All(vol.Coerce(int), vol.Range(0, 100)),
     }
 
+    def __init__(self) -> None:
+        """Initialize SetIntentHandler."""
+        super().__init__(INTENT_SET, DOMAIN, SERVICE_TURN_ON, "Turned on {}")
+        self.service_data: dict[str, Any] = {}
+        self.speech_parts: list[str] = []
+        _LOGGER.info(self.slot_schema)
+
     async def async_handle(self, intent_obj: intent.Intent) -> intent.IntentResponse:
         """Handle the hass intent."""
-        hass = intent_obj.hass
-        slots = self.async_validate_slots(intent_obj.slots)
-        state = intent.async_match_state(
-            hass, slots["name"]["value"], hass.states.async_all(DOMAIN)
-        )
+        self.service_data = {}
```

Comment:
Why is this on `self`? This means it cannot handle 2 intents in parallel

Reply:
I was contorting it to fit the mold of `ServiceIntentHandler`, so I just copied the relevant code.